From 0e100ace68e6f34606a7407c862f58a918635e6c Mon Sep 17 00:00:00 2001
From: Bastian Bittorf <bittorf@bluebottle.com>
Date: Mon, 11 Mar 2013 13:16:00 +0100
Subject: [PATCH 3/3] [packages] olsrd: init: reduce code duplication by
 reusing already_in_schema()

[packages] olsrd: init: reduce code duplication by reusing already_in_schema()

changed scope/position of this function, and reuse it.

Signed-off-by: Bastian Bittorf <bittorf@bluebottle.com>
---
 net/olsrd/files/olsrd.init |   48 +++++++++++++++++++++-----------------------
 1 file changed, 23 insertions(+), 25 deletions(-)

diff --git a/net/olsrd/files/olsrd.init b/net/olsrd/files/olsrd.init
index aab85eb..1172623 100644
--- a/net/olsrd/files/olsrd.init
+++ b/net/olsrd/files/olsrd.init
@@ -270,34 +270,32 @@ config_write_options() {
 		return 0
 	}
 
-	case " $schema " in
-		*" speed "*)
-			get_value_for_entry "speed"
+	already_in_schema()
+	{
+		case " $schema " in
+			*" $1 "*)
+				return 0
+			;;
+			*)
+				return 1
+			;;
+		esac
+	}
 
-			if [ 2>/dev/null $value -gt 0 -a $value -le 20 ]; then
-				speed="$value"
-			else
-				log "$funcname() Warning: invalid speed-value: '$value' - allowed integers: 1...20, fallback to 6"
-				speed=6
-			fi
+	already_in_schema "speed" && {
+		get_value_for_entry "speed"
 
-			already_in_schema()
-			{
-				case " $schema " in
-					*" $1 "*)
-						return 0
-					;;
-					*)
-						return 1
-					;;
-				esac
-			}
+		if [ 2>/dev/null $value -gt 0 -a $value -le 20 ]; then
+			speed="$value"
+		else
+			log "$funcname() Warning: invalid speed-value: '$value' - allowed integers: 1...20, fallback to 6"
+			speed=6
+		fi
 
-			for schema_entry in $list_speed_vars; do {
-				already_in_schema "$schema_entry" || schema="$schema $schema_entry"
-			} done
-		;;
-	esac
+		for schema_entry in $list_speed_vars; do {
+			already_in_schema "$schema_entry" || schema="$schema $schema_entry"
+		} done
+	}
 
 	for schema_entry in $schema; do
 		if [ -n "$speed" ]; then		# like sven-ola freifunk firmware fff-1.7.4
-- 
1.7.10.4

